### PR TITLE
Add SameSite

### PR DESCRIPTION
**Pull request by jackstine**
# Description

SameSite to the available CookieOptions

# Context

Users need to have the ability to add SameSite to the CookieOptions.

# Checklist:

- [X] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [X] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [X] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [X] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5123

### CODE REVIEW

File: changelog/v1.11.0-rc3/add-samesite.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: NEW_FEATURE
+    issueLink: https://github.com/solo-io/gloo/issues/5123
+    description: Add same_site to CookieOptions.
```
Comment (bewebi) [inline changelog/v1.11.0-rc3/add-samesite.yaml:4]: Based on [this roadmap](https://github.com/solo-io/gloo/issues/5123#issuecomment-1080810865) it seems that there is additional work beyond the Gloo repo that's required so we should add the
```
resolvesIssue: false
```
field

**Review by bewebi — COMMENTED** [review]
One comment on changelog

Coming at this with relatively limited context, I'm also wondering if any code change is needed to ensure desired behavior and/or if additional tests are in order. Assuming neither is needed, could you explain why?

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 2]
```diff
@@ -349,6 +349,20 @@ message UserSession {
...
+            // navigating to the origin site.
+            LaxMode = 0;
+            // Only be sent in a first-party context and not be sent along with requests
+            // initiated by third party websites.
+            StrictMode = 1;
+            // Cookies are sent in all contexts. Cookie NotSecure must be unset.
+            NoneMode = 2;
+        }
+        // Whether the cookie should be restricted to a first-party or same-site context.
+        // The default mode should be LaxMode.
```
Comment (nfuden) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:364]: Nit: change should be -> is
```suggestion
        // The default is LaxMode.
```

**Comment by gunnar-solo** [comment]
/kick [[Fail] Gateway in memory hybrid gateway http traffic ssl [It] should work with ssl if ssl config is present in matcher ](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=18568aae-6bb7-4e88-a136-849e5a029c2f)

**Comment by sam-heilbron** [comment]
/kick https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=ceb697db-5956-462f-851a-299efc8ff197

**Pull request merged**